packages/babel-generator/src/printer.js

```diff
@@ -58,7 +58,7 @@ export default class Printer extends Buffer {
     if (needsParens) this.push(")");
 
     // end
-    this.map.mark(node, "end");
+    if (parent) this.map.mark(parent);
```

Comment:
This is basically creating a stack for the source mappings. This line restores parent's mapping state so any code that it may continue to emit is mapped to the parent not this node.

It would likely be clearer in the code if mark was renamed to a push and pop pair of APIs, but I generally look to make the minimal number of changes in PRs for projects that aren't my own to avoid them becoming stuck in the review phase.